### PR TITLE
Improve documentation of PIPELINE, TRIGGER and WAIT based on user feedback

### PR DESCRIPTION
**Pull request by vladaionescu**

### CODE REVIEW

File: docs/earthfile/earthfile.md [hunk 1]
```diff
@@ -1198,8 +1198,7 @@ Even though the `WAIT` clause limits parallelism by forcing everything within it
 
 #### Examples
 
-As an example, multiple `WAIT` blocks can be used; the first block builds and pushes to a remote registry (in parallel), then a second `WAIT` block
-can be used to execute a script which requires those images to exist in the remote registry:
+As an example, a `WAIT` block can be used to build and pushe to a remote registry (in parallel), then, after that execute a script which requires those images to exist in the remote registry:
```
Comment (r-LaForge) [inline docs/earthfile/earthfile.md:1201]:
```suggestion
As an example, a `WAIT` block can be used to build and push to a remote registry (in parallel), then, after that execute a script which requires those images to exist in the remote registry:
```

**Review by r-LaForge — APPROVED** [review]

File: docs/earthfile/earthfile.md [hunk 2]
```diff
@@ -1598,12 +1595,37 @@ The `PIPELINE` command is in beta status and is only useful for Earthly CI.
 
 #### Description
 
-The `PIPELINE` command is used to declare that the current target is an Earthly CI pipeline. The `PIPELINE` command must be the first command in the target.
+The `PIPELINE` command is used to declare that the current target is an Earthly CI pipeline. The `PIPELINE` command must be the first command in the target. The `PROJECT` declaration in the Earthfile is also required, and it must match the project the repository is added to, in order for the pipeline to be executed.
```
Comment (alexcb) [inline docs/earthfile/earthfile.md:1598]: I struggle to understand `and it must match the project the repository is added to, in order for the pipeline to be executed`.

Maybe breaking it up into a separate paragraph, and rewording it to:

```suggestion
The `PIPELINE` command is used to declare that the current target is an Earthly CI pipeline. The `PIPELINE` command must be the first command in the target.

To use a `PIPELINE`, you must also declare a `PROJECT` in the Earthfile.
This `PROJECT` must match the name of the CI's project that references the git repository
```

would make it easier?

File: docs/earthfile/earthfile.md [hunk 3]
```diff
@@ -1598,12 +1595,37 @@ The `PIPELINE` command is in beta status and is only useful for Earthly CI.
...
 
 A pipeline is a target that is executed by Earthly CI when a certain trigger is activated. Triggers can be declared via the `TRIGGER` command. Pipeline targets allow only the commands `TRIGGER`, `ARG` and `BUILD`. Other commands may be used indirectly in other targets that can be then referenced by `BUILD`.
 
 Pipeline targets are always executed with no outputs, in strict mode.
 
+{% hint style='info' %}
+##### Note
+Pipelines and their definitions, including their triggers must be merged into the primary branch (by default, the default branch on GitHub) in order for the triggers to take effect.
```
Comment (alexcb) [inline docs/earthfile/earthfile.md:1606]: `by default, the default` is difficult to quickly parse, maybe we can use `overridden` for one of the `default`s instead?
```suggestion
Pipelines and their definitions, including their triggers must be merged into the primary branch (which, unless overridden, is the default branch on GitHub) in order for the triggers to take effect.
```

File: docs/earthfile/earthfile.md [hunk 4]
```diff
@@ -1633,6 +1655,11 @@ In the *PR form*, the pipeline is triggered when a pull request is opened agains
 
 In the *push form*, the pipeline is triggered when a push is made to the branch `<push-branch>`.
 
+{% hint style='info' %}
+##### Note
+Pipelines and their definitions, including their triggers must be merged into the primary branch (by default, the default branch on GitHub) in order for the triggers to take effect.
```
Comment (alexcb) [inline docs/earthfile/earthfile.md:1660]:
```suggestion
Pipelines and their definitions, including their triggers must be merged into the primary branch (which, unless overridden, is the default branch on GitHub) in order for the triggers to take effect.
```

**Review by alexcb — APPROVED** [review]
looks good, just a couple suggestions for readability.

**Pull request merged**